app/models/application.rb

```diff
@@ -0,0 +1,41 @@
+class Application < ApplicationRecord
+  validates :name, presence: true
+  validates :street_address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :zipcode, presence: true, numericality: true
+  validates :description, presence: true
+  has_many :application_pets
+
+
+  def self.application(id)
+    Application.find(id)
+  end
+
+  def full_address
+    street_address << " " << city << ", " << state << " " << zipcode
+  end
+
+  def list_of_pets
+    Pet.joins(:applications).where("application_id = ?", self.id)
+  end
+
+  def added_pets?
+    list_of_pets.present?
+  end
+
+  def all_pets_approved
+    if status_of_application_pet.uniq.count == 1 && status_of_application_pet.first
```

Comment:
The following solution would work best if you had a pet_application status that had a default value like "unreviewed" rather than `nil` but you could do something like:

```
non_approved_pets = self.application_pets.where("application_approved != 'Approved'")
return true if non_approved_pets.empty? 
false
```

This includes more AR and less Ruby